client/components/gravatar/index.jsx

```diff
@@ -33,6 +42,16 @@ module.exports = React.createClass( {
 		};
 	},
 
+	componentDidMount() {
```

Comment:
A better lifecycle method would probably be `componentWillReceiveProps` in case we update the user prop after a mount. I fwe do proceed with this, we can do an early exit if we're not rendering the current user gravatar. 

I'm still thinking a custom serialize/deserialize method would be better, and in the component we ignore the temp image if it's expired. 

It might look something like:
```javascript
import { isValidStateWithSchema } from 'state/utils';
//...
[SERIALIZE]: ( state ) => removeExpiredTempImage( state ),
[DESERIALIZE]: ( state ) => { 
if ( isValidStateWithSchema( state, schema ) ) {
    return removeExpiredTempImage( state );
}
```